app/forms/field_handlers/field_handler.py

```diff
@@ -43,11 +46,21 @@ def label(self):
     def guidance(self):
         return self.answer_schema.get("guidance", "")
 
+    @staticmethod
+    def format_question_title(error_message, question_title):
+        error_message = error_message or ""
+        if "%(question_title)s" in error_message:
+            error_message = error_message % dict(
+                question_title=safe_content(question_title)
+            )
+        return error_message
+
     def get_validation_message(self, message_key):
         message = self.answer_schema.get("validation", {}).get("messages", {}).get(
             message_key
         ) or self.error_messages.get(message_key)
-        return message
+
+        return self.format_question_title(message, self.question_title)
```

Comment:
> I don't think we need to check if there is a `question_title` placeholder in the string:

We do unfortunately. If the string contains other variables (e.g. %(min)s) and we don't provide all vars in a single call to format, it will fail with undefined vars.

Reply:
Additionally, it would make sense to use f-strings here and doing so highlights that using "min" and "max" should be avoided as they are built-in functions.

Reply:
So that means `question_title` can't be used if there are other placeholders in the string?

Reply:
Nope. Not ideal, but unless we want to come up with a dependency resolver for them I don't see how else this can be handled. Is there anywhere we'd need this behaviour?

Reply:
I don't think a dependency resolver would be needed. I think the question title could be passed down through the fields to the validators, to where the existing interpolation of min/max etc is done. As you've said though, this implementation is probably enough for what we need right now.

If we do keep this, then the schema validator should validate all error messages to check that if `question_text` exists in an error string, no other placeholders should.